packages/atomic/src/components/common/refine-modal/refine-modal-common.tsx

```diff
@@ -112,15 +111,15 @@ export function getClonedFacetElements(
   divSlot.style.flexDirection = 'column';
   divSlot.style.gap = 'var(--atomic-refine-modal-facet-margin, 20px)';
 
-  const facetElementsPayload = facetElements.map((f) => ({
-    facetId: f.getAttribute('facet-id')!,
-    payload: f,
-  }));
-  const sortedFacetsElements = facetManager
-    .sort(facetElementsPayload)
-    .map((f) => f.payload);
+  const allFacetTags = Object.keys(
+    aggregate(facetElements, (el) => el.tagName.toLowerCase())
+  );
+
+  const allFacetsInOrderInDOM = document.querySelectorAll(
+    allFacetTags.join(',')
+  );
```

Comment:
I think we should avoid using `querySelector` from `document`, in case we eventually have cases where there's multiple search interfaces in the page.

I think we should use `querySelectorAll` relative to the search interface instead.

Reply:
Yeah good point !